services/ppac/src/test/java/app/coronawarn/datadonation/services/ppac/ios/verification/PpacProcessorIntegrationTest.java

```diff
@@ -61,6 +69,77 @@ void clearDatabase() {
     when(iosDeviceApiClient.queryDeviceData(anyString(), any())).thenReturn(ResponseEntity.ok(jsonify(data)));
   }
 
+  @Test
+  public void testNewApiTokenNotSavedWhenFurtherProcessingFails() {
+    // given
+    // - a valid ApiToken that was created now.
+    String apiToken = buildUuid();
+    final OffsetDateTime now = OffsetDateTime.now(ZoneOffset.UTC);
+
+    // - a device token
+    String deviceToken = buildBase64String(this.configuration.getIos().getMinDeviceTokenLength() + 1);
+    // - a data submission payload
+    PPADataRequestIOS submissionPayloadIos = buildPPADataRequestIosPayload(apiToken, deviceToken, false);
+
+    // when
+    // - checking the device token then return some valid mock data.
+    final PerDeviceDataResponse perDeviceDataResponse = buildIosDeviceData(now.minusMonths(1), true);
+    when(iosDeviceApiClient.queryDeviceData(anyString(), any()))
+        .thenReturn(ResponseEntity.ok(jsonify(perDeviceDataResponse)));
+    // - when updating the device data then fail
+    when(iosDeviceApiClient.updatePerDeviceData(anyString(), any()))
+        .thenThrow(FeignException.class);
+    // then
+    // - failed submission
+    final ResponseEntity<DataSubmissionResponse> dataSubmissionResponseResponseEntity = postSubmission(
+        submissionPayloadIos, testRestTemplate,
+        IOS_SERVICE_URL, true);
```

Comment:
variable `dataSubmissionResponseResponseEntity` isn't used at all... 

```suggestion
    postSubmission(submissionPayloadIos, testRestTemplate, IOS_SERVICE_URL, true);
```